lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -956,17 +971,21 @@ defmodule Mix.Tasks.Xref do
           {roots, callback, count} =
             roots_and_callback(file_references, filter, sources, sinks, opts)
 
-          path = Keyword.get(opts, :output, "xref_graph.dot")
+          file_spec = Mix.Utils.write_dot_graph!("xref graph", Enum.sort(roots), callback, opts)
 
-          Mix.Utils.write_dot_graph!(path, "xref graph", Enum.sort(roots), callback, opts)
+          if file_spec != "-" do
+            png_file_spec = (file_spec |> Path.rootname() |> Path.basename()) <> ".png"
 
-          if path != "-" do
-            png_path = (path |> Path.rootname() |> Path.basename()) <> ".png"
+            location =
+              if Path.dirname(file_spec) == "." and !String.starts_with?(file_spec, "."),
+                do: " in the current directory",
+                else: ""
 
+            # maybe this would be better as a comment in the dot file.
```

Comment:
We can probably remove this comment. I would keep the experience like this as people may not know .dot is a text file.

```suggestion
```

Reply:
Ok, fair enough.